api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -759,26 +885,41 @@ public int hashCode() {
 
   public static class MapType extends NestedType {
     public static MapType ofOptional(int keyId, int valueId, Type keyType, Type valueType) {
+      return ofOptional(keyId, valueId, keyType, valueType, null);
+    }
+
+    public static MapType ofOptional(int keyId, int valueId, Type keyType, Type valueType,
+        Map<Object, Object> defaultValue) {
```

Comment:
I am confused why we need to pass a default value here. Are not the parameters of this method already at the level of the Map keys and values? Passing a Map-level concept seems incompatible with what this method is trying to achieve.

Reply:
Also, this method (while new) is not called anywhere in this patch.

Reply:
that is true. Removed defaultValue from Map/List/Struct level, it is sufficient to maintain it at the NestedField level.